CONTRIBUTING.md

```diff
@@ -1,49 +1,115 @@
+After your PR is reviewed, make any necessary changes by repeating steps 2 and 3.
+A team member will merge your PR after approval,
+ensuring it's reviewed by someone who has not contributed to the branch.
 
-## Branches
+### 2. Creating Feature #2 (Developer #2) and Rebasing
+
+#### Step 1: Update Local Main Branch
+Pull and rebase the remote `main` onto your local `main` branch to keep it up to date.
+```bash
+git pull --rebase origin main
+```
 
-Here's a brief description of the supported branches in our repository:
+#### Step 2: Develop New Functionality
+Add new functionality commits to your branch.
 
-![Release Deployment workflow](images/release-deployment.png)
+#### Step 3: Push Changes
+Push your code up to GitHub.
 
-| Branch           | Protected? | Base Branch        | Description                                                                                                                                                                                        |
-|:-----------------|:-----------|:-------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `main`           | YES        | N/A                | What is live in production (**stable**).<br/>A pull request is required to merge code into `main`.                                                                                                 |
-| `dev`            | YES        | `main`             | The latest state of development (**unstable**).<br/>A pull request is required to merge code into `dev`                                                                                            |
-| `feature/*`      | NO         | `dev`              | Cutting-edge features (**unstable**). These branches are used for any maintenance features / active development.                                                                                   |
-| `release/vX.Y.Z` | NO         | `dev`              | A temporary release branch that follows the [semver](http://semver.org/) versioning. This is what is sent to UAT.<br/>A pull request is required to merge code into any `release/vX.Y.Z` branch.   |
-| `hotfix/*`       | NO         | `main`             | These are bug fixes against production.<br/>This is used because develop might have moved on from the last published state.<br/>Remember to merge this back into develop and any release branches. |
+#### Step 4: Perform Tests
+Ensure to perform any necessary tests to validate the new functionality.
 
+#### Step 5: Open a PR
+Open a PR for review. Note that PR may show merge conflicts with `main` due to concurrent feature merges.
+
+#### Step 6: Rebasing
+First, update your local `main` branch.
+```bash
+git checkout main
+git pull --rebase origin main
+```
+Then, rebase your feature branch with `main` to integrate the latest changes.
+```bash
+git checkout <feature-2-branch>
```

Comment:
Would mention after checkout, you will need to stash in your changes by executing:

```
git stash
```

Then do git stash apply{0}

Reply:
I don't think you need to do that accordingly to this logic as your changes have already been committed to the branch you are checking out. Then you rebase against the updated main you pulled with your dev branch.